app/Services/Manager.php

```diff
@@ -322,36 +323,92 @@ public function appendReportback($data, $parameters)
         $places = ['1st', '2nd', '3rd'];
         $prizeCopy = ['Taking home the $100 AMEX gift card', 'Winner of the $50 amex gift card', 'Winner of the $25 amex gift card'];
 
         $topThree = [];
+        $includeUserIds = false;
+        $hasCompetitionId = false;
+        $hasMessageId = false;
 
-        foreach ($topThreeUsers as $key => $user) {
-            $reportbackItems = $user->reportback->reportback_items->data;
-            $latestReportbackItem = array_pop($reportbackItems);
+        if (isset($options['includeUserIds']) && $options['includeUserIds']) {
+            $includeUserIds = true;
+        }
+
+        if (isset($options['competition_id']) && isset($options['message_id'])) {
+            $hasCompetitionId = true;
+            $hasMessageId = true;
+        }
 
+        foreach ($topThreeUsers as $key => $user) {
+            // Basic info on top three
             $topThree[] = [
-                'place' => $places[$key],
-                'prize_copy' => $prizeCopy[$key],
-                'first_name' => $user->first_name,
-                'quantity' => $user->reportback->quantity,
-                'image_url' => $latestReportbackItem->media->uri,
-                'caption' => $latestReportbackItem->caption,
+              'place' => $places[$key],
+              'first_name' => $user->first_name,
+              'prize_copy' => $prizeCopy[$key],
+              'quantity' => $user->reportback->quantity,
             ];
+
+            // Provide info on user & reportback ids
+            if ($includeUserIds) {
+                $topThree[$key]['user_id'] = $user->id;
+                $topThree[$key]['reportback_id'] = $user->reportback->id;
+            }
+
+            // Provide image url/captaion of top three leaderboard images
+            if ($hasCompetitionId && $hasMessageId) {
+                $competition_id = $options['competition_id'];
+                $message_id = $options['message_id'];
+
+                $leaderboardReportbackItem = $this->getLeaderboardPhoto($competition_id, $message_id, $user->id);   //@NOTE calling Phoenix again
+
+                if (! isset($leaderboardReportbackItem)) {
+                    $reportbackItems = $user->reportback->reportback_items->data;
+                    $leaderboardReportbackItem = array_pop($reportbackItems);
+                }
+
+                $topThree[$key]['image_url'] = $leaderboardReportbackItem->media->uri;
+                $topThree[$key]['caption'] = $leaderboardReportbackItem->caption;
+            }
         }
 
         return $topThree;
     }
 
+    /*
+     * Determines whether a valid reportback_item _id is present in the LeaderboardPhotos DB
+     * for given competition, message, and leaderboard user.
+     * Returns the associated reportback item if present.
+     *
+     * @return  array $reportback_item/null
+     */
+    public function getLeaderboardPhoto($competition_id, $message_id, $user_id)
```

Comment:
These params should be 🐫  case (`$competitionId`). it's a laravel convention. variables/params are  🐫 case, database column are 🐍 case